controllers/rollingupgrade_controller.go

```diff
@@ -139,6 +140,14 @@ func (r *RollingUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{RequeueAfter: v1alpha1.DefaultRequeueTime}, nil
 	}
 
+	// only reconcile maxParallel number of CRs at a time
+	if common.GetSyncMapLen(&r.AdmissionMap) >= r.maxParallel {
+		if _, present := r.AdmissionMap.Load(rollingUpgrade.NamespacedName()); !present {
+			r.Info("number of ongoing CRs has reached the max-parallel limit, requeuing", "name", rollingUpgrade.NamespacedName())
```

Comment:
@ZihanJiang96 did you get a chance to test this on a large cluster? By enforcing this limit, we might slow down upgrades significantly. 

Here's what we could do:
- Create a dev cluster with 100 IGs. (maybe 2 nodes in each.)
- Do a cluster upgrade with current version of upgrade-manager (with maxUnavailable set to 100%)
    - Monitor the memory & CPU consumption.
    - Measure how much time it took, overall.
- Do a cluster upgrade with your changes. (with maxUnavailable set to 100%)
    - Monitor the memory & CPU consumption.
    - Measure how much time it took, overall.